analysis/org.eclipse.tracecompass.analysis.timing.core/src/org/eclipse/tracecompass/internal/analysis/timing/core/segmentstore/SegmentStoreStatisticsDataProvider.java

```diff
@@ -32,57 +29,42 @@
 import org.eclipse.tracecompass.segmentstore.core.ISegment;
 import org.eclipse.tracecompass.tmf.core.analysis.IAnalysisModule;
 import org.eclipse.tracecompass.tmf.core.dataprovider.DataProviderParameterUtils;
-import org.eclipse.tracecompass.tmf.core.model.AbstractTmfTraceDataProvider;
 import org.eclipse.tracecompass.tmf.core.model.CommonStatusMessage;
-import org.eclipse.tracecompass.tmf.core.model.ITableColumnDescriptor;
 import org.eclipse.tracecompass.tmf.core.model.filters.FilterTimeQueryFilter;
 import org.eclipse.tracecompass.tmf.core.model.filters.TimeQueryFilter;
-import org.eclipse.tracecompass.tmf.core.model.tree.ITmfTreeDataProvider;
 import org.eclipse.tracecompass.tmf.core.model.tree.TmfTreeModel;
 import org.eclipse.tracecompass.tmf.core.response.ITmfResponse.Status;
 import org.eclipse.tracecompass.tmf.core.response.TmfModelResponse;
 import org.eclipse.tracecompass.tmf.core.trace.ITmfTrace;
 
-import com.google.common.collect.ImmutableList;
-
 /**
- * This data provider will supply a statistics tree for an
- * {@link AbstractSegmentStatisticsAnalysis}. Passing a
- * {@link FilterTimeQueryFilter} will also return the statistics from the
- * selected range.
+ * This data provider delivers a statistics tree for an
+ * {@link AbstractSegmentStatisticsAnalysis} or any analysis module implementing
+ * {@link IStatisticsAnalysis}, featuring a fixed two-level structure. By
+ * passing a {@link FilterTimeQueryFilter}, it also returns statistics for the
+ * specified range.
  *
  * @author Loic Prieur-Drevon
  * @since 4.0
  */
-public class SegmentStoreStatisticsDataProvider extends AbstractTmfTraceDataProvider
-        implements ITmfTreeDataProvider<SegmentStoreStatisticsModel> {
+public class SegmentStoreStatisticsDataProvider extends AbstractSegmentStoreStatisticsDataProvider {
 
     /**
      * Base {@link SegmentStoreStatisticsDataProvider} prefix
      */
     public static final String ID = "org.eclipse.tracecompass.analysis.timing.core.segmentstore.SegmentStoreStatisticsDataProvider"; //$NON-NLS-1$
 
     private static final String STATISTICS_SUFFIX = ".statistics"; //$NON-NLS-1$
-    private static final String TOTAL_PREFIX = "Total_"; //$NON-NLS-1$
-    private static final String SELECTION_PREFIX = "Selection_"; //$NON-NLS-1$
     private static final Map<IStatisticsAnalysis, SegmentStoreStatisticsDataProvider> PROVIDER_MAP = new WeakHashMap<>();
-    private static final AtomicLong ENTRY_ID = new AtomicLong();
 
     private final IStatisticsAnalysis fProvider;
-    private final String fId;
-    private @Nullable String fRootEntryName;
-
-    private final Map<String, Long> fIdToType = new HashMap<>();
-    private final long fTraceId = ENTRY_ID.getAndIncrement();
-
     private final @Nullable IAnalysisModule fModule;
 
-    private SegmentStoreStatisticsAspects fAspects = new SegmentStoreStatisticsAspects();
+    private final Map<String, Long> fIdToType = new HashMap<>();
```

Comment:
remove. This map was moved to the abstract class and is not needed here. The dispose() method will have to be updated for that.